modules/compute_metric.py

```diff
@@ -0,0 +1,143 @@
+import torch.distributed as dist
+
+from monai.data import create_test_image_3d, partition_dataset
+from monai.handlers import write_metrics_reports
+from monai.metrics import DiceMetric
+from monai.transforms import (
+    AsDiscreted,
+    EnsureChannelFirstd,
+    Compose,
+    KeepLargestConnectedComponentd,
+    LoadImaged,
+    ScaleIntensityd,
+    ToTensord,
+)
+from monai.utils import string_list_all_gather
+
+
+def compute(args):
+    # generate synthetic data for the example
+    if args.local_rank == 0 and not os.path.exists(args.dir):
+        # create 16 random pred, label paris for evaluation
+        print(f"generating synthetic data to {args.dir} (this may take a while)")
+        os.makedirs(args.dir)
+        # if have multiple nodes, set random seed to generate same random data for every node
+        np.random.seed(seed=0)
+        for i in range(16):
+            pred, label = create_test_image_3d(128, 128, 128, num_seg_classes=1, channel_dim=-1, noise_max=0.5)
+            n = nib.Nifti1Image(pred, np.eye(4))
+            nib.save(n, os.path.join(args.dir, f"pred{i:d}.nii.gz"))
+            n = nib.Nifti1Image(label, np.eye(4))
+            nib.save(n, os.path.join(args.dir, f"label{i:d}.nii.gz"))
+
+    # initialize the distributed evaluation process, change to NCCL backend if computing on GPU
+    dist.init_process_group(backend="gloo", init_method="env://")
+
+    preds = sorted(glob(os.path.join(args.dir, "pred*.nii.gz")))
+    labels = sorted(glob(os.path.join(args.dir, "label*.nii.gz")))
+    datalist = [{"pred": pred, "label": label} for pred, label in zip(preds, labels)]
+
+    # split data for every subprocess, for example, 16 processes compute in parallel
+    data_part = partition_dataset(
+        data=datalist,
+        num_partitions=dist.get_world_size(),
+        shuffle=False,
+        even_divisible=False,
+    )[dist.get_rank()]
+
+    # define transforms for predictions and labels
+    transforms = Compose(
+        [
+            LoadImaged(keys=["pred", "label"]),
+            EnsureChannelFirstd(keys=["pred", "label"]),
+            ScaleIntensityd(keys="pred"),
+            ToTensord(keys=["pred", "label"]),
+            AsDiscreted(keys="pred", threshold_values=True),
+            KeepLargestConnectedComponentd(keys="pred", applied_labels=[1]),
+        ]
+    )
+    data_part = [transforms(item) for item in data_part]
+
+    # compute metrics for current process
+    metric = DiceMetric(include_background=True, reduction="mean", get_not_nans=False)
+    metric(y_pred=[i["pred"] for i in data_part], y=[i["label"] for i in data_part])
+    filenames = [item["pred_meta_dict"]["filename_or_obj"] for item in data_part]
+    # all-gather results from all the processes and reduce for final result
+    result = metric.aggregate().item()
+    filenames = string_list_all_gather(strings=filenames)
+
+    if args.local_rank == 0:
+        print("mean dice: ", result)
+        # generate metric report
```

Comment:
`# generate metrics at output/mean_dice_raw.csv, output/mean_dice_summary.csv, output/metrics.csv`

Reply:
updated in the latest commit.

Thanks.